tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -1286,76 +1311,153 @@ class ExecutionTest {
     return hr;
   }
 
-  static HRESULT
-  EnableExperimentalShaderModels(bool bExperimentalShaderModels = true,
-                                 UINT SDKVersion = 0, LPCSTR SDKPath = "") {
-    HMODULE hRuntime = LoadLibraryW(L"d3d12.dll");
-    if (hRuntime == NULL) {
-      return HRESULT_FROM_WIN32(GetLastError());
+  static std::wstring GetModuleName() {
+    wchar_t moduleName[MAX_PATH+1] = {0};
+    DWORD length = GetModuleFileNameW(NULL, moduleName, MAX_PATH);
+    if (length == 0 || length == MAX_PATH) {
+      return std::wstring(); // Error condition
     }
+    return std::wstring(moduleName, length);
+  }
+
+  static std::wstring ComputeSDKFullPath(std::wstring SDKPath) {
+    std::wstring modulePath = GetModuleName();
+    size_t pos = modulePath.rfind('\\');
+    if (pos == std::wstring::npos)
+      return SDKPath;
+    if (SDKPath.substr(0, 2) != L".\\")
+      return SDKPath;
+    return modulePath.substr(0, pos) + SDKPath.substr(1);
+  }
 
-    if (SDKVersion) {
-      D3D12GetInterfaceFn pD3D12GetInterface =
-          (D3D12GetInterfaceFn)GetProcAddress(hRuntime, "D3D12GetInterface");
-      CComPtr<ID3D12SDKConfiguration> pD3D12SDKConfiguration;
-      IFR(pD3D12GetInterface(CLSID_D3D12SDKConfiguration, IID_PPV_ARGS(&pD3D12SDKConfiguration)));
-      IFR(pD3D12SDKConfiguration->SetSDKVersion(SDKVersion, SDKPath));
+  static UINT GetD3D12SDKVersion(std::wstring SDKPath) {
+    // Try to automatically get the D3D12SDKVersion from the DLL
+    UINT SDKVersion = 0;
+    std::wstring D3DCorePath = ComputeSDKFullPath(SDKPath);
+    D3DCorePath.append(L"D3D12Core.dll");
+    HMODULE hCore = LoadLibraryW(D3DCorePath.c_str());
+    if (hCore) {
+      if (UINT *pSDKVersion = (UINT*)GetProcAddress(hCore, "D3D12SDKVersion"))
+        SDKVersion = *pSDKVersion;
```

Comment:
Missing () after *pSDKVersion?

Reply:
The export is a UINT*, not a function to be called.

Reply:
Oh I see :)